data/virt_autotest/virt_logs_collector.sh

```diff
@@ -344,6 +344,51 @@ function collect_extra_logs_from_host() {
 	return ${ret_result}
 }
 
+#This function supports collecting guest configure xml file.
+#Typical usage: collect_guest_config logs_folder guest
+function collect_guest_config() {
+        local logs_folder=$1
+        local guest_name=$2
+        local guest_config="${logs_folder}/${guest_name}"
+
+        #Collect guest configure xml file
+        echo -e "Collect ${guest_name} configure xml file"
+        virsh dumpxml ${guest_name} | tee ${guest_config}-config.xml
+        if [[ $? -eq 0 ]];then
```

Comment:
$? flags the return value of 'tee' instead of 'virsh' in most bash if I remember correctly ~ so the if loop is meaningless here.

Reply:
good point, will confirm it as your mentioned.

Reply:
Quick update, as the default , the return status of a pipeline is the exit status of the last (rightmost) command in bash script. 
So, we will keep the same return status between of `collect_guest_autoyast ${virt_logs_folder} ${guest_current} | tee -a ${virt_logs_collecor_log}` and `collect_guest_autoyast() `together. 
we'd better to use with **pipefail**:
```
If pipefail is enabled, the pipeline's return status is the value of the last (rightmost) command 
to exit with a non-zero status, or zero if all commands exit successfully. 
If the reserved word ! precedes a pipeline,  the exit  status of that pipeline is the logical 
negation of  the exit status as described above.  
The shell waits for all commands in the pipeline to terminate before  return‐ing a value.
```

Also, I had confirmed **pipefail** manually in bash script:

```
# cat foo.sh 
#!/bin/bash
set -o pipefail

function test() {

    #DEBUG bash with pipeline
    false | true
        
}

test
```

```

# bash -x foo.sh
+ set -o pipefail
+ test
+ false
+ true
vh001:/tmp # echo $?
1
```


```
# cat foo.sh 
#!/bin/bash
set -o pipefail

function test() {

    #DEBUG bash with pipeline
    true | false
        
}

test
```


```
# bash -x foo.sh
+ set -o pipefail
+ test
+ true
+ false
vh001:/tmp # echo $?
1
```
So far, had added `set -o pipefail` to virt_logs_collector.sh bash script.
Also, add the latest validation run:
[gi-guest_sles15sp2-on-host_developing-kvm](http://149.44.176.58/tests/5722523#)

Reply:
Quick update,  confirmed that the return status of a pipeline is the exit status of the last (rightmost) command in bash script as the default.
```
vh001:~ # bash -x foo.sh 
+ test
+ false
+ ret=1
+ echo 1
+ true
vh001:~ # echo $?
0
vh001:~ # cat foo.sh 
#!/bin/bash

function test() {

#DEBUG $? with pipeline
false
ret=$?
echo "$ret" | true

}

test
```